### ISSUE TITLE
Third-Party XB1 controller

### ISSUE THREAD

**Issue by J2Kbr** [J2Kbr/TitanTwo#224]
```
[USB DESCRIPTOR]
12 01 00 02 FF FF FF 40 C6 24 1A 58 01 01 01 02 03 01 09 02 60 00 03 01 00 A0 FA 09 04 00 00 02 FF 47 D0 00 07 05 02 03 40 00 04 07 05 81 03 40 00 04 09 04 01 00 00 FF 47 D0 00 09 04 01 01 02 FF 47 D0 00 07 05 08 01 E4 00 01 07 05 87 01 E4 00 01 09 04 02 00 00 FF 47 D0 00 09 04 02 01 02 FF 47 D0 00 07 05 04 02 40 00 00 07 05 83 02 40 00 00 
[STRING 1 DESCRIPTOR]
00 00 00 02 00 00 02 01 80 06 03 01 04 09 00 08 
08 03 42 00 44 00 41 00 
[STRING 2 DESCRIPTOR]
00 00 00 02 00 00 02 01 80 06 03 02 04 09 00 2E 
2E 03 58 00 42 00 31 00 20 00 43 00 6C 00 61 00 73 00 73 00 69 00 63 00 20 00 43 00 6F 00 6E 00 74 00 72 00 6F 00 6C 00 6C 00 65 00 72 00 
[STRING 3 DESCRIPTOR]
00 00 00 02 00 00 02 01 80 06 03 03 04 09 00 22 
22 03 30 00 30 00 30 00 30 00 31 00 41 00 33 00 44 00 43 00 35 00 46 00 39 00 37 00 32 00 32 00 32 00 
[USB SNIFF]
00 00 00 01 00 00 01 00 80 06 01 00 00 00 00 40 
00 00 00 02 00 00 02 01 80 06 01 00 00 00 00 40 
12 01 00 02 FF FF FF 40 C6 24 1A 58 01 01 01 02 03 01 
00 00 00 01 00 00 01 00 00 05 00 1D 00 00 00 00 
00 00 00 02 00 00 02 01 00 05 00 03 00 00 00 00 
1D 
00 00 00 01 00 00 01 00 80 06 01 00 00 00 00 12 
00 00 00 02 00 00 02 01 80 06 01 00 00 00 00 12 
12 01 00 02 FF FF FF 40 C6 24 1A 58 01 01 01 02 03 01 
00 00 00 01 00 00 01 00 80 06 02 00 00 00 00 FF 
00 00 00 02 00 00 02 01 80 06 02 00 00 00 00 FF 
09 02 60 00 03 01 00 A0 FA 09 04 00 00 02 FF 47 D0 00 07 05 02 03 40 00 04 07 05 81 03 40 00 04 09 04 01 00 00 FF 47 D0 00 09 04 01 01 02 FF 47 D0 00 07 05 08 01 E4 00 01 07 05 87 01 E4 00 01 09 04 02 00 00 FF 47 D0 00 09 04 02 01 02 FF 47 D0 00 07 05 04 02 40 00 00 07 05 83 02 40 00 00 
00 00 00 01 00 00 01 00 80 06 03 EE 00 00 00 12 
00 00 00 02 00 00 02 01 80 06 03 EE 00 00 00 12 
12 03 4D 00 53 00 46 00 54 00 31 00 30 00 30 00 90 00 
00 00 00 01 00 00 01 00 80 06 03 03 04 09 00 FF 
00 00 00 02 00 00 02 01 80 06 03 03 04 09 00 FF 
22 03 30 00 30 00 30 00 30 00 31 00 41 00 33 00 44 00 43 00 35 00 46 00 39 00 37 00 32 00 32 00 32 00 
00 00 00 01 00 00 01 00 C0 90 00 00 00 04 00 10 
00 00 00 02 00 00 02 01 C0 90 00 00 00 04 00 10 
28 00 00 00 00 01 04 00 01 00 00 00 00 00 00 00 
00 00 00 01 00 00 01 00 C0 90 00 00 00 04 00 28 
00 00 00 02 00 00 02 01 C0 90 00 00 00 04 00 28 
28 00 00 00 00 01 04 00 01 00 00 00 00 00 00 00 00 01 58 47 49 50 31 30 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
00 00 00 01 00 00 01 00 80 06 03 00 00 00 00 FF 
00 00 00 02 00 00 02 01 80 06 03 00 00 00 00 FF 
04 03 09 04 
00 00 00 01 00 00 01 00 80 06 03 02 04 09 00 FF 
00 00 00 02 00 00 02 01 80 06 03 02 04 09 00 FF 
2E 03 58 00 42 00 31 00 20 00 43 00 6C 00 61 00 73 00 73 00 69 00 63 00 20 00 43 00 6F 00 6E 00 74 00 72 00 6F 00 6C 00 6C 00 65 00 72 00 
00 00 00 01 00 00 01 00 80 06 06 00 00 00 00 0A 
00 00 00 02 00 00 02 02 80 06 06 00 00 00 00 0A 
00 00 00 01 00 00 01 00 C1 90 00 00 00 05 00 0A 
00 00 00 02 00 00 02 02 C1 90 00 00 00 05 00 0A 
00 00 00 01 00 00 01 00 80 06 01 00 00 00 00 12 
00 00 00 02 00 00 02 01 80 06 01 00 00 00 00 12 
12 01 00 02 FF FF FF 40 C6 24 1A 58 01 01 01 02 03 01 
00 00 00 01 00 00 01 00 80 06 02 00 00 00 00 09 
00 00 00 02 00 00 02 01 80 06 02 00 00 00 00 09 
09 02 60 00 03 01 00 A0 FA 
00 00 00 01 00 00 01 00 80 06 02 00 00 00 00 60 
00 00 00 02 00 00 02 01 80 06 02 00 00 00 00 60 
09 02 60 00 03 01 00 A0 FA 09 04 00 00 02 FF 47 D0 00 07 05 02 03 40 00 04 07 05 81 03 40 00 04 09 04 01 00 00 FF 47 D0 00 09 04 01 01 02 FF 47 D0 00 07 05 08 01 E4 00 01 07 05 87 01 E4 00 01 09 04 02 00 00 FF 47 D0 00 09 04 02 01 02 FF 47 D0 00 07 05 04 02 40 00 00 07 05 83 02 40 00 00 
00 00 00 01 00 00 01 00 80 00 00 00 00 00 00 02 
00 00 00 02 00 00 02 01 80 00 00 00 00 00 00 02 
02 00 
00 00 00 01 00 00 01 00 00 09 00 01 00 00 00 00 
00 00 00 02 00 00 02 01 00 09 00 01 00 00 00 00 
00 00 00 01 00 00 01 00 01 0B 00 00 00 01 00 00 
00 00 00 02 00 00 02 01 01 0B 00 00 00 01 00 00 
00 00 00 01 00 00 01 00 01 0B 00 00 00 02 00 00 
00 00 00 02 00 00 02 01 01 0B 00 00 00 02 00 00 
00 00 00 02 00 01 
02 20 01 1C 48 E8 93 E2 34 A1 00 00 C6 24 1A 58 01 00 01 00 17 01 02 00 01 00 01 00 01 00 01 00 
00 00 00 01 00 00 01 00 80 06 01 00 00 00 00 40 
00 00 00 02 00 00 02 01 80 06 01 00 00 00 00 40 
12 01 00 02 FF FF FF 40 C6 24 1A 58 01 01 01 02 03 01 
00 00 00 01 00 00 01 00 00 05 00 1D 00 00 00 00 
00 00 00 02 00 00 02 01 00 05 00 03 00 00 00 00 
1D 
00 00 00 01 00 00 01 00 80 06 01 00 00 00 00 12 
00 00 00 02 00 00 02 01 80 06 01 00 00 00 00 12 
12 01 00 02 FF FF FF 40 C6 24 1A 58 01 01 01 02 03 01 
00 00 00 01 00 00 01 00 00 09 00 01 00 00 00 00 
00 00 00 02 00 00 02 01 00 09 00 01 00 00 00 00 
00 00 00 01 00 00 01 00 01 0B 00 00 00 01 00 00 
00 00 00 02 00 00 02 01 01 0B 00 00 00 01 00 00 
00 00 00 01 00 00 01 00 01 0B 00 00 00 02 00 00 
00 00 00 02 00 00 02 01 01 0B 00 00 00 02 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 02 00 01 
02 20 03 1C 48 E8 93 E2 34 A1 00 00 C6 24 1A 58 01 00 01 00 17 01 02 00 01 00 01 00 01 00 01 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 
00 00 00 01 00 00 01 00 02 01 00 00 00 81 00 00 
00 00 00 02 00 00 02 01 02 01 00 00 00 81 00 00 

```

**Comment by J2Kbr** [comment]
Message sent to the user:

> I am working on the support for this controller and, after checking the provided descriptors, it seems everything is already supported and should work with the Titan Two.
> 
> Could you please inform how this controller behaves when plugged into the Titan Two?
> 
> -	Does the status LED of the USB port turns green? red? remains off?
> -	Can you notice any button activity on Gtuner’s Device Monitor?
> 
> Any other information you can provide will be very helpful.

**Comment by J2Kbr** [comment]
Reply from user:

> Turns green on the titan two but doesn't work device monitor says nothing

![3](https://user-images.githubusercontent.com/21229433/63360654-21e00500-c345-11e9-8a05-c457e8af77a0.jpg)
![1](https://user-images.githubusercontent.com/21229433/63360660-23a9c880-c345-11e9-80ac-79e556d41f5f.jpg)

**Comment by J2Kbr** [comment]
User informed the controller works when in NOT passthru mode.
![2](https://user-images.githubusercontent.com/21229433/63367570-83f33700-c352-11e9-8719-fd323a8b2dc4.jpg)

**Comment by J2Kbr** [comment]
Possible fix with firmware 1.04
